index.js

```diff
@@ -46,8 +46,10 @@ NefitEasyAccessory.prototype.getTemperature = function(type, prop, callback) {
     return this.client.status(true);
   }).then((status) => {
     var temp = status[prop];
-    this.log.debug('...%s temperature is %s', type, temp);
-    return callback(null, temp);
+    if (!isNaN(temp) && isFinite(temp)) {
+      this.log.debug('...%s temperature is %s', type, temp);
+      return callback(null, temp);
```

Comment:
Are you sure it's okay to never call the callback in case the value isn't valid?

Reply:
You are right, just thought about that too. I will update it soon.

Reply:
If I understand `hap-nodejs` correctly, it's valid to return a `null` value for more characteristics, so this might work: 

```
return callback(null, (! isNan(temp) && isFinite(temp)) ? temp : null);
```

Reply:
Thank you for the suggestion, I have tested it with my HomeBridge and Nefit Easy and got these warnings:

`[homebridge-nefit-easy] This plugin generated a warning from the characteristic 'Current Temperature': characteristic was supplied illegal value: null! Home App will reject null for Apple defined characteristics. See https://homebridge.io/w/JtMGR for more info.
[homebridge-nefit-easy] This plugin generated a warning from the characteristic 'Target Temperature': characteristic was supplied illegal value: null! Home App will reject null for Apple defined characteristics. See https://homebridge.io/w/JtMGR for more info.`

So I did it this way:

`if (!isNaN(temp) && isFinite(temp)) {
      this.log.debug('...%s temperature is %s', type, temp);
      return callback(null, temp);
    }
    else {
      return callback(new Error("Device returned NaN as temperature value."));
    }`
    
    It is tested and works fine for me.